src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -846,35 +846,46 @@ public async ValueTask<HttpResponseMessage> SendWithRetryAsync(HttpRequestMessag
                         response = await connection!.SendAsync(request, async, cancellationToken).ConfigureAwait(false);
                     }
                 }
-                catch (HttpRequestException e) when (e.AllowRetry == RequestRetryType.RetryOnLowerHttpVersion)
+                catch (HttpRequestException e) when (e.AllowRetry == RequestRetryType.RetryOnConnectionFailure)
                 {
-                    // Throw since fallback is not allowed by the version policy.
-                    if (request.VersionPolicy != HttpVersionPolicy.RequestVersionOrLower)
+                    retryCount++;
+                    if (retryCount == MaxConnectionFailureRetries)
```

Comment:
Nitpicky, but it seems like MaxConnectionFailureRetries is actually MaxConnectionFailureTries?  i.e. it's the total number of attempts we'll make, including the initial, yes?  This is only meaningful because if we ever changed MaxConnectionFailureRetries to 0 in the future (or made it user-configurable) thinking "I don't want any retries, just the initial attempt", this would end up essentially being an infinite retry loop.

Reply:
Yes, that's a good point. I will change this to actually be MaxConnectionFailureRetries, i.e. make it 4 and rework the logic here.